src/components/TextInput/BaseTextInput.js

```diff
@@ -287,6 +287,8 @@ class BaseTextInput extends Component {
                                         keyboardType={getSecureEntryKeyboardType(this.props.keyboardType, this.props.secureTextEntry, this.state.passwordHidden)}
                                         value={this.state.value}
                                         selection={this.state.selection}
+                                        dataSet={{submitOnEnter: this.props.multiline && this.props.submitOnEnter}}
```

Comment:
>         // FormSubmit Enter key handler does not have access to direct props.
        // `dataset.submitOnEnter` is used to indicate that pressing Enter on this input should call the submit callback.
        
        
        This comment was meant to be used here.